### ISSUE TITLE
empty while statement doesn't account for conditions with side-effects

### ISSUE THREAD

**Issue by LegalizeAdulthood** [oclint/oclint#235]
Run oclint on [assimp](https://github.com/LegalizeAdulthood/assimp/blob/master/code/3DSLoader.cpp#L719) and it reports empty while statements on code like this:

```
while (stream->GetI1());
```

The [documentation for empty while loop check](http://docs.oclint.org/en/dev/rules/empty.html#emptywhilestatement) says that it detects a while loop that does nothing.  The code only checks for a while loop with no statements, not "doing nothing", because it doesn't check the condition for side-effects.  This check will give many false positives on real-world code that has loops of the variety of "loop with side effects until this condition is no longer true"

**Comment by ryuichis** [comment]
Understood, please consider suppressing this line by appending `//!OCLINT`. Or disable this rule entirely.
